src/LanguageServer/Protocol/Handler/BufferedProgress.cs

```diff
@@ -43,14 +43,17 @@ public void Dispose()
         /// </summary>
         public void Report(T value)
         {
-            // Don't need to lock _underlyingProgress.  It is inherently thread-safe itself being an IProgress implementation.
-            _underlyingProgress?.Report(value);
-
-            if (_buffer != null)
+            lock (_buffer)
```

Comment:
This will null-ref if `_buffer` is null, which is checked below. What's the purpose?

Reply:
this PR is not for review.  it's sharing between me and david as i isolate which part of the stack has the issue.